R/utilities.R

```diff
@@ -313,13 +318,85 @@ bind_scores.default <- function(x) {
 
 #' @noRd
 #' @export
+#'
+#' @examples
+#' # Create a list of `score_obj`
+#' # Bind scores
+#' score_obj_list <- list(score_obj_aov, score_obj_cor, score_obj_imp)
+#' score_obj_list |> bind_scores()
+#'
 bind_scores.list <- function(x) {
   score_set <- x[[1]]$score_res
   for (i in 2:length(x)) {
-    score_set <- dplyr::full_join(score_set, x[[i]]$score_res)
+    score_set <- dplyr::full_join(
+      score_set,
+      x[[i]]$score_res,
+      by = c("name", "score", "outcome", "predictor") # OR suppressMessages()
+    )
   }
   score_set <- score_set |>
     tidyr::pivot_wider(names_from = name, values_from = score)
   class(score_set) <- c("score_set", class(score_set))
   score_set
 }
+
+#' Fill in safe values.
+#'
+#' @param x A list where each element is a score object of class `score_obj`.
+#'
+#' @export
+fill_safe_values <- function(x) {
+  UseMethod("fill_safe_values")
+}
+
+#' @noRd
+#' @export
+fill_safe_values.default <- function(x) {
+  cli::cli_abort(
+    "{.arg x} must be {.cls list}, not {.obj_type_friendly {x}}."
+  )
+}
+
+#' @noRd
+#' @export
+#'
+#' @examples
+#' # Create a list of `score_obj`
+#' # Fill in safe values
+#' score_obj_list <- list(score_obj_aov, score_obj_cor, score_obj_imp)
+#' score_obj_list |> fill_safe_values()
+#'
+fill_safe_values.list <- function(x) {
+  score_set <- x[[1]]$score_res
+  for (i in 2:length(x)) {
+    score_set <- dplyr::full_join(
+      score_set,
+      x[[i]]$score_res,
+      by = c("name", "score", "outcome", "predictor") # OR suppressMessages()
+    )
+  }
+
+  for (i in 1:length(x)) {
+    # TODO Wonder if there is a cleaner way to do it. map?
+    method_name <- unique(x[[i]]$score_res$name)
```

Comment:
You already have the value in the object

```suggestion
    method_name <- x[[i]]$score_type
```